warcio/recordloader.py

```diff
@@ -124,20 +131,28 @@ def parse_record_stream(self, stream,
             length = 0
 
         # limit stream to the length for all valid records
+        check_payload_digest = False
         if length is not None and length >= 0:
             stream = LimitReader.wrap_stream(stream, length)
-
```

Comment:
Maybe this could just be:
```
  if check_digests:
     stream, is_verifying = make_digest_verifying_stream(stream)

def wrap_digest_verifying_stream(self, stream):
     try:
         stream = DigestVerifyingReader(stream, record_type=rec_type,
                                                               payload_digest=rec_headers.get_header('WARC-Payload-Digest')
                                                               block_digest=rec_headers.get_header('WARC-Block-Digest')
                                                               segment_number=rec_headers.get_header('WARC-Segment-Number'))
         return stream, True
      except:
         return stream, False